test/TestUtil.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+const assert = require('assert');
+const print = require('../lib/util/print');
+const flatten = require('../lib/util/flattenOperands');
+const math = require('mathjs');
+
+// Test helper contains methods for reducing the redundant code. (Test Util)
+const TestUtil = {};
+
+// Test the distribute steps in the expression string.
+TestUtil.testDistributeSteps = function (evaluator, exprString, outputList) {
+  const lastString = outputList[outputList.length - 1];
+  it(exprString + ' -> ' + lastString, () => {
+    const status = evaluator(flatten(math.parse(exprString)));
+    const substeps = status.substeps;
+
+    assert.deepEqual(substeps.length, outputList.length);
+    substeps.forEach((step, i) => {
+      assert.deepEqual(
+        print(step.newNode),
+        outputList[i]);
+    });
+
+    assert.deepEqual(
+      print(status.newNode),
+      lastString);
+  });
+};
+
+// Test the prime factors and the factor pairs
+TestUtil.testFactors = function (fn, input, output) {
```

Comment:
you could rename this to be more general, since it's serving a very general purpose

maybe `TestUtil.testFunctionOutput`?